src/entitysdk/downloaders/ion_channel_model.py

```diff
@@ -0,0 +1,22 @@
+"""Download functions for IonChannelModel entities."""
+
+from pathlib import Path
+
+
+def download_one_mechanism(client, ic, mechanisms_dir="./mechanisms"):
+    """Download one mechanism file.
+
+    Args:
+        client (Client): EntitySDK client
+        ic (IonChannelModel): IonChannelModel entitysdk object
+        mechanisms_dir (str or Pathlib.Path): directory to save the mechanism file
+    """
+    mechanisms_dir = Path(mechanisms_dir)
+    mechanisms_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Given that you use this pattern in many places it can be factored out and be put in 
```
src/entitysdk/utils/filesystem.py
```
```python
StrOrPath = str | os.PathLike[str]  # this can go in entitysdk.types

def create_dir(path: StrOrPath) -> Path:
    """Create directory and parents if it doesn't already exist."""
    path = Path(path)
    path.mkdir(parents=True, exist_ok=True)
    return path
```
And then you can use it directly in
```python
output_path=create_dir(output_dir)
```